src/main/java/gyro/google/dns/PolicyResource.java

```diff
@@ -0,0 +1,276 @@
+
+    @Override
+    public boolean doRefresh() throws Exception {
+        Dns client = createClient(Dns.class);
+        Policy policy = client.policies().get(getProjectId(), getName()).execute();
+        return refreshFrom(policy);
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state) throws Exception {
+        Dns client = createClient(Dns.class);
+        Policy policy = new Policy();
+        DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig = getAlternativeNameServerConfig();
+
+        if (alternativeNameServerConfig != null) {
+            policy.setAlternativeNameServerConfig(alternativeNameServerConfig.copyTo());
+        }
+        policy.setDescription(getDescription());
+        policy.setEnableInboundForwarding(getEnableInboundForwarding());
+        policy.setEnableLogging(getEnableLogging());
+        policy.setName(getName());
+        List<DnsPolicyNetwork> network = getNetwork();
+
+        if (network != null) {
+            policy.setNetworks(network
+                .stream()
+                .map(DnsPolicyNetwork::copyTo)
+                .collect(Collectors.toList()));
+        }
+        Policy response = client.policies().create(getProjectId(), policy).execute();
+        refreshFrom(response);
+    }
+
+    @Override
+    public void doUpdate(
+        GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        if (!(current instanceof PolicyResource)) {
+            throw new GyroException("Incompatible resource type! " + current.getClass().getName());
+        }
+        Policy policy = new Policy();
+
+        for (String changedFieldName : changedFieldNames) {
+            if (changedFieldName.equals("description")) {
+                policy.setDescription(getDescription());
+            } else if (changedFieldName.equals("alternative-name-server-config")) {
+                DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig = getAlternativeNameServerConfig();
+                policy.setAlternativeNameServerConfig(alternativeNameServerConfig == null
+                    ? Data.nullOf(PolicyAlternativeNameServerConfig.class)
+                    : alternativeNameServerConfig.copyTo());
+            } else if (changedFieldName.equals("enable-inbound-forwarding")) {
+                policy.setEnableInboundForwarding(getEnableInboundForwarding());
+            } else if (changedFieldName.equals("enable-logging")) {
+                policy.setEnableLogging(getEnableLogging());
+            } else if (changedFieldName.equals("network")) {
+                List<DnsPolicyNetwork> networks = getNetwork();
+                policy.setNetworks(networks == null
+                    ? Data.nullOf(ArrayList.class)
+                    : networks.stream()
+                        .map(DnsPolicyNetwork::copyTo)
+                        .collect(Collectors.toList()));
+            }
+        }
+        Dns client = createClient(Dns.class);
```

Comment:
Conventionally we do declare the client at the top.

Reply:
I think declaring variables closer to where they are used is better as it's easier to understand and organize, and there's a chance that variables never get used if they are declared too early.
Also if you run through a some kind of linter like sonar, it will complain that it is declared too early.
so I rather change the convention to move variables close to where they are being used.